### ISSUE TITLE
Add AppSync perms to lambda function

### ISSUE THREAD

**Issue by aprilmintacpineda** [aws-amplify/amplify-cli#6675]
>**Note**: If your feature-request is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

**Is your feature request related to a problem? Please describe.**

I am accessing the graphql api this way:

```js
const AWSAppSyncClient = require('aws-appsync').default;

// on the handler

const client = new AWSAppSyncClient({
  url: process.env.API_WRGDGRAPHQLAPI_GRAPHQLAPIENDPOINTOUTPUT,
  region: process.env.REGION,
  disableOffline: true,
  auth: {
    type: 'AWS_IAM',
    credentials: AWS.config.credentials
  }
});
```

Then I can do:

```js
client.mutate({ mutation, variables });
```

or 

```js
client.query({ quer, variables });
```

But the problem is, if I add the `graphql api` to the resource access list of the lambda function, it doesn't add the other necessary appsync permissions:

```json
{
  "Effect": "Allow",
  "Action": [
    "appsync:GraphQL"
  ],
```

But it needs to be:

```json
{
  "Effect": "Allow",
  "Action": [
    "appsync:Create*",
    "appsync:StartSchemaCreation",
    "appsync:GraphQL",
    "appsync:Get*",
    "appsync:List*",
    "appsync:Update*",
    "appsync:Delete*"
  ],
```

So I have to manually add this, which is a VERY BAD THING because when I do `update function` and change the api access, this will be rewritten, the best way is for amplify to offer a way to add these perms because amplify itself is modifying these files.

**Describe the solution you'd like**

Something like `amplify update function` then `AppSync` is one of the options OR add these perms by default when I give a function `API` access.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

**Comment by wderezin** [comment]
The IAM role permission that amplify cli generate has changed.  I have confirm this was changed as early as `4.41.2` and confirmed it is still broke in `4.44.0` 

Currently the `amplify cli` generate function api permissions in `-cloudformation-template.json` as: 
```json
 {
              "Effect": "Allow",
              "Action": [
                "appsync:GraphQL"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:appsync:",
                      {
                        "Ref": "AWS::Region"
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":apis/",
                      {
                        "Ref": "apiknotGraphQLAPIIdOutput"
                      },
                      "/types/create/*"
                    ]
                  ]
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:appsync:",
                      {
                        "Ref": "AWS::Region"
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":apis/",
                      {
                        "Ref": "apiknotGraphQLAPIIdOutput"
                      },
                      "/types/read/*"
                    ]
                  ]
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:appsync:",
                      {
                        "Ref": "AWS::Region"
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":apis/",
                      {
                        "Ref": "apiknotGraphQLAPIIdOutput"
                      },
                      "/types/update/*"
                    ]
                  ]
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:appsync:",
                      {
                        "Ref": "AWS::Region"
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":apis/",
                      {
                        "Ref": "apiknotGraphQLAPIIdOutput"
                      },
                      "/types/delete/*"
                    ]
                  ]
                }
              ]
            },
```

Previous `amplfiy cli` versions generated:
```json
  {
              "Effect": "Allow",
              "Action": [
                "appsync:Create*",
                "appsync:StartSchemaCreation",
                "appsync:GraphQL",
                "appsync:Get*",
                "appsync:List*",
                "appsync:Update*",
                "appsync:Delete*"
              ],
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:appsync:",
                      {
                        "Ref": "AWS::Region"
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":apis/",
                      {
                        "Ref": "apiknotGraphQLAPIIdOutput"
                      },
                      "/*"
                    ]
                  ]
                }
              ]
            }
```

**Comment by thibaultdalban** [comment]
I think this issue is a duplicate of #6595

**Comment by wderezin** [comment]
@thibaultdalban yes this is a duplicate.  I'll add my comment over there.

**Comment by aprilmintacpineda** [comment]
Thanks guys, hopefully gets fixed soon 🙇🏻

**Comment by r0zar** [comment]
@attilah do you have bandwidth to tackle this?

**Comment by SwaySway** [comment]
Providing an update here

It looks like it's generating the old policy format if the feature flag is not set to true. If you set the feature flag to true and update the flow again it will use the new graphql operation list as opposed to the crud list. Put up a pr to maintain the FF use when creating the iam policy for appsync.

PR #6273

**Comment by anothrNick** [comment]
@SwaySway The PR linked above is another issue. Is there an open PR that resolves this AppSync issue?

**Comment by kylekirkby** [comment]
Any idea when this is going to be fixed? This was working fine before... @attilah @SwaySway What needs to be done to be able to access graphql in a lambda function again..? I'm not really wanting to add crazy custom cloud formation template workarounds :/

**Comment by renegoretzka** [comment]
I am running into the same issue. Hopefully the PR gets merged asap.

**Comment by kylekirkby** [comment]
@renegoretzka I added a custom amplify resource to amend the lambda execution policies to unblock myself. I too hope they fix this soon! It was working perfectly in previous versions.

**Comment by renegoretzka** [comment]
> @renegoretzka I added a custom amplify resource to amend the lambda execution policies to unblock myself. I too hope they fix this soon! It was working perfectly in previous versions.

How did you solve it exactly? I also would like to continue translating my App from the AppSync API to DataStore. @kylekirkby

**Comment by kylekirkby** [comment]
@renegoretzka 

It's a bit hacky but it's a workaround 🤷‍♂️

https://serverlessstartup.hashnode.dev/amending-aws-amplify-lambda-permissions-with-a-custom-cloudformation-template

**Comment by fkunecke** [comment]
Still seems to be an issue. @SwaySway's solution here fixed this for me: https://github.com/aws-amplify/amplify-cli/issues/6595#issuecomment-777755187

I'm on 4.45.2. Before adding `generateGraphQLPermissions` I was getting the following in my CF template:
```
{
  "Effect": "Allow",
  "Action": [
    "appsync:GraphQL"
  ],
  "Resource": [
    {
      "Fn::Join": [
        "",
        [
          "arn:aws:appsync:",
          {
            "Ref": "AWS::Region"
          },
          ":",
          {
            "Ref": "AWS::AccountId"
          },
          ":apis/",
          {
            "Ref": "apiturnspeedGraphQLAPIIdOutput"
          },
          "/types/create/*"
        ]
      ]
    },
    {
      "Fn::Join": [
        "",
        [
          "arn:aws:appsync:",
          {
            "Ref": "AWS::Region"
          },
          ":",
          {
            "Ref": "AWS::AccountId"
          },
          ":apis/",
          {
            "Ref": "apiturnspeedGraphQLAPIIdOutput"
          },
          "/types/read/*"
        ]
      ]
    },
    {
      "Fn::Join": [
        "",
        [
          "arn:aws:appsync:",
          {
            "Ref": "AWS::Region"
          },
          ":",
          {
            "Ref": "AWS::AccountId"
          },
          ":apis/",
          {
            "Ref": "apiturnspeedGraphQLAPIIdOutput"
          },
          "/types/update/*"
        ]
      ]
    },
    {
      "Fn::Join": [
        "",
        [
          "arn:aws:appsync:",
          {
            "Ref": "AWS::Region"
          },
          ":",
          {
            "Ref": "AWS::AccountId"
          },
          ":apis/",
          {
            "Ref": "apiturnspeedGraphQLAPIIdOutput"
          },
          "/types/delete/*"
        ]
      ]
    }
  ]
}
```
